{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[84]\n"
     ]
    }
   ],
   "source": [
    "!cd datasets/std && python create.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+ model_name=AE_skipLSTM_AR\n",
      "+ seed=0\n",
      "+ windows=6\n",
      "+ train_percent=0.1\n",
      "+ dataset=std\n",
      "+ e_init=10\n",
      "+ e_train=5\n",
      "+ e_reinit=100\n",
      "+ n_cpd=10\n",
      "+ unit=20\n",
      "+ horizon=4\n",
      "+ highway=6\n",
      "+ skip_sizes='3 5 7'\n",
      "+ python3 -W ignore ./code/main.py --dataset_name std --seed 0 --windows 6 --train_percent 0.1 --model_name AE_skipLSTM_AR --unit 20 --horizon 4 --highway 6 --skip_sizes 3 5 7 --e_init 10 --e_train 5 --e_reinit 100 --n_cpd 10\n",
      "Namespace(dataset_name='std', e_init=10, e_reinit=100, e_train=5, highway=6, horizon=4, model_name='AE_skipLSTM_AR', n_cpd=10, seed=0, skip_sizes=[3, 5, 7], train_percent=0.1, unit=20, windows=6)\n",
      "{'dataset_name': 'std', 'skip_sizes': [3, 5, 7], 'train_percent': 0.1, 'training_epochs': 10, 'horizon': 4, 'highway': 6, 'model_names': ['AE_skipLSTM_AR', 'AE_skipLSTM_AR', 'AE_skipLSTM_AR'], 'lr': 0.001, 'GRUUnits': 20, 'SkipGRUUnits': 20, 'window': 6, 'num_change_threshold': 7, 'num_ano_cpd': 10, 'threshold_high': 4, 'threshold_low': 1.4, 'epochs_to_train_after_cpd': 100, 'extra_samples_after_cpd': 3, 'epochs_to_train_single_sample': 5, 'file_name': './results_AE_skipLSTM_AR/std/seed=0/', 'ensemble_space': 3}\n",
      "WARNING:tensorflow:From ./code/main.py:40: The name tf.random.set_random_seed is deprecated. Please use tf.compat.v1.random.set_random_seed instead.\n",
      "\n",
      "./datasets/std/std.json\n",
      "Log path  ./results_AE_skipLSTM_AR/std/seed=0/log.out\n",
      "2025-03-12 00:55:00.716659: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcuda.so.1\n",
      "2025-03-12 00:55:00.747898: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:55:00.748258: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: \n",
      "name: NVIDIA GeForce GTX 1650 major: 7 minor: 5 memoryClockRate(GHz): 1.515\n",
      "pciBusID: 0000:01:00.0\n",
      "2025-03-12 00:55:00.749286: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
      "2025-03-12 00:55:00.750879: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
      "2025-03-12 00:55:00.752036: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0\n",
      "2025-03-12 00:55:00.754494: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0\n",
      "2025-03-12 00:55:00.758204: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0\n",
      "2025-03-12 00:55:00.760193: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0\n",
      "2025-03-12 00:55:00.767964: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7\n",
      "2025-03-12 00:55:00.768446: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:55:00.769353: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:55:00.770116: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0\n",
      "2025-03-12 00:55:00.771570: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA\n",
      "2025-03-12 00:55:00.786684: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2499950000 Hz\n",
      "2025-03-12 00:55:00.789952: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55eca38547b0 executing computations on platform Host. Devices:\n",
      "2025-03-12 00:55:00.790024: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): <undefined>, <undefined>\n",
      "2025-03-12 00:55:00.962682: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:55:00.963453: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55eca4819fe0 executing computations on platform CUDA. Devices:\n",
      "2025-03-12 00:55:00.963526: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): NVIDIA GeForce GTX 1650, Compute Capability 7.5\n",
      "2025-03-12 00:55:00.964529: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:55:00.964709: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: \n",
      "name: NVIDIA GeForce GTX 1650 major: 7 minor: 5 memoryClockRate(GHz): 1.515\n",
      "pciBusID: 0000:01:00.0\n",
      "2025-03-12 00:55:00.964990: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
      "2025-03-12 00:55:00.965031: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
      "2025-03-12 00:55:00.965062: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0\n",
      "2025-03-12 00:55:00.965083: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0\n",
      "2025-03-12 00:55:00.965179: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0\n",
      "2025-03-12 00:55:00.965215: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0\n",
      "2025-03-12 00:55:00.965240: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7\n",
      "2025-03-12 00:55:00.965600: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:55:00.965938: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:55:00.966097: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0\n",
      "2025-03-12 00:55:00.966377: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
      "2025-03-12 00:55:00.966878: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1181] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
      "2025-03-12 00:55:00.966897: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1187]      0 \n",
      "2025-03-12 00:55:00.966908: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 0:   N \n",
      "2025-03-12 00:55:00.967063: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:55:00.967300: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:55:00.967450: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1326] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 3545 MB memory) -> physical GPU (device: 0, name: NVIDIA GeForce GTX 1650, pci bus id: 0000:01:00.0, compute capability: 7.5)\n",
      "WARNING:tensorflow:From /home/himanshu/anaconda3/envs/BTP/lib/python3.6/site-packages/tensorflow/python/ops/init_ops.py:1251: calling VarianceScaling.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f09dfeae400>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f09dfeae400>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f09dc0f7e48>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f09dc0f7e48>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f09dc0f14e0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f09dc0f14e0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f09dc0d4048>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f09dc0d4048>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f09dc0882b0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f09dc0882b0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f09dc097908>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f09dc097908>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f09d7fc6780>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f09d7fc6780>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f09d7fe4400>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f09d7fe4400>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f09d7ffdc18>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f09d7ffdc18>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f09d7f9a240>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f09d7f9a240>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f09d7f74ba8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f09d7f74ba8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f09d7f705c0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f09d7f705c0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:From ./code/model_util.py:98: The name tf.global_variables_initializer is deprecated. Please use tf.compat.v1.global_variables_initializer instead.\n",
      "\n",
      "WARNING:tensorflow:From /home/himanshu/anaconda3/envs/BTP/lib/python3.6/site-packages/tensorflow/python/ops/math_grad.py:1250: add_dispatch_support.<locals>.wrapper (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Use tf.where in 2.0, which has the same broadcast rule as np.where\n",
      "2025-03-12 00:55:09.404868: W tensorflow/compiler/jit/mark_for_compilation_pass.cc:1412] (One-time warning): Not using XLA:CPU for cluster because envvar TF_XLA_FLAGS=--tf_xla_cpu_global_jit was not set.  If you want XLA:CPU, either set that envvar, or use experimental_jit_scope to enable XLA:CPU.  To confirm that XLA is active, pass --vmodule=xla_compilation_cache=1 (as a proper command-line flag, not via TF_XLA_FLAGS) or set the envvar XLA_FLAGS=--xla_hlo_profile.\n",
      "2025-03-12 00:55:09.519980: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
      "2025-03-12 00:55:11.398538: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:55:11.398771: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: \n",
      "name: NVIDIA GeForce GTX 1650 major: 7 minor: 5 memoryClockRate(GHz): 1.515\n",
      "pciBusID: 0000:01:00.0\n",
      "2025-03-12 00:55:11.398879: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
      "2025-03-12 00:55:11.398908: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
      "2025-03-12 00:55:11.398921: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0\n",
      "2025-03-12 00:55:11.398932: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0\n",
      "2025-03-12 00:55:11.398942: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0\n",
      "2025-03-12 00:55:11.398952: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0\n",
      "2025-03-12 00:55:11.398963: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7\n",
      "2025-03-12 00:55:11.399068: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:55:11.399193: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:55:11.399265: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0\n",
      "2025-03-12 00:55:11.399296: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1181] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
      "2025-03-12 00:55:11.399301: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1187]      0 \n",
      "2025-03-12 00:55:11.399306: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 0:   N \n",
      "2025-03-12 00:55:11.399383: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:55:11.399519: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:55:11.399612: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1326] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 3545 MB memory) -> physical GPU (device: 0, name: NVIDIA GeForce GTX 1650, pci bus id: 0000:01:00.0, compute capability: 7.5)\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f092c27c748>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f092c27c748>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f092c23a3c8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f092c23a3c8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f092c233940>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f092c233940>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f092c20f0b8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f092c20f0b8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f092c1c3390>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f092c1c3390>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f092c1604e0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f092c1604e0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f092c17f860>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f092c17f860>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f092c1349e8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f092c1349e8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f092c0d5cf8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f092c0d5cf8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f092c0f1cc0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f092c0f1cc0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f092c0b55f8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f092c0b55f8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f092c0c8be0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f092c0c8be0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "2025-03-12 00:55:17.492189: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:55:17.492314: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: \n",
      "name: NVIDIA GeForce GTX 1650 major: 7 minor: 5 memoryClockRate(GHz): 1.515\n",
      "pciBusID: 0000:01:00.0\n",
      "2025-03-12 00:55:17.492356: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
      "2025-03-12 00:55:17.492368: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
      "2025-03-12 00:55:17.492378: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0\n",
      "2025-03-12 00:55:17.492388: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0\n",
      "2025-03-12 00:55:17.492398: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0\n",
      "2025-03-12 00:55:17.492407: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0\n",
      "2025-03-12 00:55:17.492418: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7\n",
      "2025-03-12 00:55:17.492475: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:55:17.492577: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:55:17.492643: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0\n",
      "2025-03-12 00:55:17.492661: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1181] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
      "2025-03-12 00:55:17.492665: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1187]      0 \n",
      "2025-03-12 00:55:17.492669: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 0:   N \n",
      "2025-03-12 00:55:17.492725: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:55:17.492825: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:55:17.492899: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1326] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 3545 MB memory) -> physical GPU (device: 0, name: NVIDIA GeForce GTX 1650, pci bus id: 0000:01:00.0, compute capability: 7.5)\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f081e073d30>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f081e073d30>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f081e024d30>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f081e024d30>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f092c209668>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f092c209668>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f081dfe9a90>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f081dfe9a90>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f081dfbc0b8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f081dfbc0b8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f081df3fb00>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f081df3fb00>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f081df0ef60>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f081df0ef60>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f081df261d0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f081df261d0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f081dee5860>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f081dee5860>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f081de8f1d0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f081de8f1d0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f081deabda0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f081deabda0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f081de46cf8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f081de46cf8>>: AttributeError: module 'gast' has no attribute 'Index'\n"
     ]
    }
   ],
   "source": [
    "!bash run.sbatch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 0.21658471928380585 13 0.18159319941181562\n",
      "7.246425\n",
      "Figure(640x480)\n",
      "84\n"
     ]
    }
   ],
   "source": [
    "!python exact.py"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![alacpd](./results_AE_skipLSTM_AR/std/seed=0/log/1.png)\n",
    "![exact](./exact_fig/1.png)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[135]\n"
     ]
    }
   ],
   "source": [
    "!cd datasets/std && python create.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+ model_name=AE_skipLSTM_AR\n",
      "+ seed=0\n",
      "+ windows=6\n",
      "+ train_percent=0.1\n",
      "+ dataset=std\n",
      "+ e_init=10\n",
      "+ e_train=5\n",
      "+ e_reinit=100\n",
      "+ n_cpd=10\n",
      "+ unit=20\n",
      "+ horizon=4\n",
      "+ highway=6\n",
      "+ skip_sizes='3 5 7'\n",
      "+ python3 -W ignore ./code/main.py --dataset_name std --seed 0 --windows 6 --train_percent 0.1 --model_name AE_skipLSTM_AR --unit 20 --horizon 4 --highway 6 --skip_sizes 3 5 7 --e_init 10 --e_train 5 --e_reinit 100 --n_cpd 10\n",
      "Namespace(dataset_name='std', e_init=10, e_reinit=100, e_train=5, highway=6, horizon=4, model_name='AE_skipLSTM_AR', n_cpd=10, seed=0, skip_sizes=[3, 5, 7], train_percent=0.1, unit=20, windows=6)\n",
      "{'dataset_name': 'std', 'skip_sizes': [3, 5, 7], 'train_percent': 0.1, 'training_epochs': 10, 'horizon': 4, 'highway': 6, 'model_names': ['AE_skipLSTM_AR', 'AE_skipLSTM_AR', 'AE_skipLSTM_AR'], 'lr': 0.001, 'GRUUnits': 20, 'SkipGRUUnits': 20, 'window': 6, 'num_change_threshold': 7, 'num_ano_cpd': 10, 'threshold_high': 4, 'threshold_low': 1.4, 'epochs_to_train_after_cpd': 100, 'extra_samples_after_cpd': 3, 'epochs_to_train_single_sample': 5, 'file_name': './results_AE_skipLSTM_AR/std/seed=0/', 'ensemble_space': 3}\n",
      "WARNING:tensorflow:From ./code/main.py:40: The name tf.random.set_random_seed is deprecated. Please use tf.compat.v1.random.set_random_seed instead.\n",
      "\n",
      "./datasets/std/std.json\n",
      "Log path  ./results_AE_skipLSTM_AR/std/seed=0/log.out\n",
      "2025-03-12 00:58:47.416712: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcuda.so.1\n",
      "2025-03-12 00:58:47.441058: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:58:47.441240: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: \n",
      "name: NVIDIA GeForce GTX 1650 major: 7 minor: 5 memoryClockRate(GHz): 1.515\n",
      "pciBusID: 0000:01:00.0\n",
      "2025-03-12 00:58:47.441457: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
      "2025-03-12 00:58:47.444823: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
      "2025-03-12 00:58:47.451744: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0\n",
      "2025-03-12 00:58:47.453082: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0\n",
      "2025-03-12 00:58:47.456294: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0\n",
      "2025-03-12 00:58:47.457691: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0\n",
      "2025-03-12 00:58:47.467758: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7\n",
      "2025-03-12 00:58:47.467878: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:58:47.468076: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:58:47.468214: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0\n",
      "2025-03-12 00:58:47.468453: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA\n",
      "2025-03-12 00:58:47.475144: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2499950000 Hz\n",
      "2025-03-12 00:58:47.475704: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x556424221660 executing computations on platform Host. Devices:\n",
      "2025-03-12 00:58:47.475724: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): <undefined>, <undefined>\n",
      "2025-03-12 00:58:47.536039: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:58:47.536474: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x5564254ebe30 executing computations on platform CUDA. Devices:\n",
      "2025-03-12 00:58:47.536532: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): NVIDIA GeForce GTX 1650, Compute Capability 7.5\n",
      "2025-03-12 00:58:47.536999: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:58:47.537299: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: \n",
      "name: NVIDIA GeForce GTX 1650 major: 7 minor: 5 memoryClockRate(GHz): 1.515\n",
      "pciBusID: 0000:01:00.0\n",
      "2025-03-12 00:58:47.537362: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
      "2025-03-12 00:58:47.537392: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
      "2025-03-12 00:58:47.537418: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0\n",
      "2025-03-12 00:58:47.537443: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0\n",
      "2025-03-12 00:58:47.537470: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0\n",
      "2025-03-12 00:58:47.537496: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0\n",
      "2025-03-12 00:58:47.537523: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7\n",
      "2025-03-12 00:58:47.537642: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:58:47.537899: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:58:47.538063: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0\n",
      "2025-03-12 00:58:47.538120: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
      "2025-03-12 00:58:47.538717: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1181] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
      "2025-03-12 00:58:47.538741: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1187]      0 \n",
      "2025-03-12 00:58:47.538757: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 0:   N \n",
      "2025-03-12 00:58:47.538926: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:58:47.539196: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:58:47.539394: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1326] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 3545 MB memory) -> physical GPU (device: 0, name: NVIDIA GeForce GTX 1650, pci bus id: 0000:01:00.0, compute capability: 7.5)\n",
      "WARNING:tensorflow:From /home/himanshu/anaconda3/envs/BTP/lib/python3.6/site-packages/tensorflow/python/ops/init_ops.py:1251: calling VarianceScaling.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc62016f908>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc62016f908>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc6200de2e8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc6200de2e8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc6200d1390>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc6200d1390>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc6200b5390>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc6200b5390>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc620063ef0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc620063ef0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc620079780>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc620079780>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc620130b70>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc620130b70>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc60ffe2240>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc60ffe2240>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc60fffcc18>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc60fffcc18>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc60ff9ac50>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc60ff9ac50>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc60ff74978>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc60ff74978>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc60ff6ffd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc60ff6ffd0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:From ./code/model_util.py:98: The name tf.global_variables_initializer is deprecated. Please use tf.compat.v1.global_variables_initializer instead.\n",
      "\n",
      "WARNING:tensorflow:From /home/himanshu/anaconda3/envs/BTP/lib/python3.6/site-packages/tensorflow/python/ops/math_grad.py:1250: add_dispatch_support.<locals>.wrapper (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Use tf.where in 2.0, which has the same broadcast rule as np.where\n",
      "2025-03-12 00:58:50.584151: W tensorflow/compiler/jit/mark_for_compilation_pass.cc:1412] (One-time warning): Not using XLA:CPU for cluster because envvar TF_XLA_FLAGS=--tf_xla_cpu_global_jit was not set.  If you want XLA:CPU, either set that envvar, or use experimental_jit_scope to enable XLA:CPU.  To confirm that XLA is active, pass --vmodule=xla_compilation_cache=1 (as a proper command-line flag, not via TF_XLA_FLAGS) or set the envvar XLA_FLAGS=--xla_hlo_profile.\n",
      "2025-03-12 00:58:50.652802: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
      "2025-03-12 00:58:51.965351: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:58:51.965467: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: \n",
      "name: NVIDIA GeForce GTX 1650 major: 7 minor: 5 memoryClockRate(GHz): 1.515\n",
      "pciBusID: 0000:01:00.0\n",
      "2025-03-12 00:58:51.965505: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
      "2025-03-12 00:58:51.965517: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
      "2025-03-12 00:58:51.965527: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0\n",
      "2025-03-12 00:58:51.965537: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0\n",
      "2025-03-12 00:58:51.965546: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0\n",
      "2025-03-12 00:58:51.965556: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0\n",
      "2025-03-12 00:58:51.965566: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7\n",
      "2025-03-12 00:58:51.965621: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:58:51.965721: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:58:51.965786: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0\n",
      "2025-03-12 00:58:51.965802: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1181] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
      "2025-03-12 00:58:51.965807: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1187]      0 \n",
      "2025-03-12 00:58:51.965811: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 0:   N \n",
      "2025-03-12 00:58:51.965866: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:58:51.965965: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:58:51.966039: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1326] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 3545 MB memory) -> physical GPU (device: 0, name: NVIDIA GeForce GTX 1650, pci bus id: 0000:01:00.0, compute capability: 7.5)\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc5841a34a8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc5841a34a8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc584158390>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc584158390>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc584156390>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc584156390>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc584135f28>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc584135f28>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc5840e8320>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc5840e8320>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc5840f8908>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc5840f8908>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc5840a6710>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc5840a6710>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc58405a780>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc58405a780>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc5687bbf98>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc5687bbf98>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc5687d6898>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc5687d6898>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc568779160>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc568779160>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc5687afd68>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc5687afd68>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "2025-03-12 00:58:55.482990: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:58:55.483107: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: \n",
      "name: NVIDIA GeForce GTX 1650 major: 7 minor: 5 memoryClockRate(GHz): 1.515\n",
      "pciBusID: 0000:01:00.0\n",
      "2025-03-12 00:58:55.483146: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
      "2025-03-12 00:58:55.483158: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
      "2025-03-12 00:58:55.483168: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0\n",
      "2025-03-12 00:58:55.483178: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0\n",
      "2025-03-12 00:58:55.483188: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0\n",
      "2025-03-12 00:58:55.483198: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0\n",
      "2025-03-12 00:58:55.483209: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7\n",
      "2025-03-12 00:58:55.483265: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:58:55.483364: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:58:55.483430: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0\n",
      "2025-03-12 00:58:55.483447: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1181] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
      "2025-03-12 00:58:55.483451: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1187]      0 \n",
      "2025-03-12 00:58:55.483455: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 0:   N \n",
      "2025-03-12 00:58:55.483510: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:58:55.483611: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 00:58:55.483683: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1326] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 3545 MB memory) -> physical GPU (device: 0, name: NVIDIA GeForce GTX 1650, pci bus id: 0000:01:00.0, compute capability: 7.5)\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc54c34ad68>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc54c34ad68>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc54c2fcdd8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc54c2fcdd8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc584113ba8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc584113ba8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc54c2c1518>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc54c2c1518>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc54c293978>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc54c293978>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc54c2174a8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc54c2174a8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc54c1e5668>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc54c1e5668>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc54c1fe2e8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc54c1fe2e8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc54c1bc6d8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc54c1bc6d8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc54c162748>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc54c162748>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc54c182ef0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7fc54c182ef0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc54c11dd30>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7fc54c11dd30>>: AttributeError: module 'gast' has no attribute 'Index'\n"
     ]
    }
   ],
   "source": [
    "!bash run.sbatch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16 0.9410324671210936 25 0.14846166562699914\n",
      "16.459653\n",
      "Figure(640x480)\n",
      "135\n"
     ]
    }
   ],
   "source": [
    "!python exact.py"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![alacpd](./results_AE_skipLSTM_AR/std/seed=0/log/2.png)\n",
    "![exact](./exact_fig/2.png)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[148]\n"
     ]
    }
   ],
   "source": [
    "!cd datasets/std && python create.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+ model_name=AE_skipLSTM_AR\n",
      "+ seed=0\n",
      "+ windows=6\n",
      "+ train_percent=0.1\n",
      "+ dataset=std\n",
      "+ e_init=10\n",
      "+ e_train=5\n",
      "+ e_reinit=100\n",
      "+ n_cpd=10\n",
      "+ unit=20\n",
      "+ horizon=4\n",
      "+ highway=6\n",
      "+ skip_sizes='3 5 7'\n",
      "+ python3 -W ignore ./code/main.py --dataset_name std --seed 0 --windows 6 --train_percent 0.1 --model_name AE_skipLSTM_AR --unit 20 --horizon 4 --highway 6 --skip_sizes 3 5 7 --e_init 10 --e_train 5 --e_reinit 100 --n_cpd 10\n",
      "Namespace(dataset_name='std', e_init=10, e_reinit=100, e_train=5, highway=6, horizon=4, model_name='AE_skipLSTM_AR', n_cpd=10, seed=0, skip_sizes=[3, 5, 7], train_percent=0.1, unit=20, windows=6)\n",
      "{'dataset_name': 'std', 'skip_sizes': [3, 5, 7], 'train_percent': 0.1, 'training_epochs': 10, 'horizon': 4, 'highway': 6, 'model_names': ['AE_skipLSTM_AR', 'AE_skipLSTM_AR', 'AE_skipLSTM_AR'], 'lr': 0.001, 'GRUUnits': 20, 'SkipGRUUnits': 20, 'window': 6, 'num_change_threshold': 7, 'num_ano_cpd': 10, 'threshold_high': 4, 'threshold_low': 1.4, 'epochs_to_train_after_cpd': 100, 'extra_samples_after_cpd': 3, 'epochs_to_train_single_sample': 5, 'file_name': './results_AE_skipLSTM_AR/std/seed=0/', 'ensemble_space': 3}\n",
      "WARNING:tensorflow:From ./code/main.py:40: The name tf.random.set_random_seed is deprecated. Please use tf.compat.v1.random.set_random_seed instead.\n",
      "\n",
      "./datasets/std/std.json\n",
      "Log path  ./results_AE_skipLSTM_AR/std/seed=0/log.out\n",
      "2025-03-12 01:03:39.272654: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcuda.so.1\n",
      "2025-03-12 01:03:39.303643: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:03:39.303961: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: \n",
      "name: NVIDIA GeForce GTX 1650 major: 7 minor: 5 memoryClockRate(GHz): 1.515\n",
      "pciBusID: 0000:01:00.0\n",
      "2025-03-12 01:03:39.305238: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
      "2025-03-12 01:03:39.307668: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
      "2025-03-12 01:03:39.309330: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0\n",
      "2025-03-12 01:03:39.310710: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0\n",
      "2025-03-12 01:03:39.314700: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0\n",
      "2025-03-12 01:03:39.318344: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0\n",
      "2025-03-12 01:03:39.327590: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7\n",
      "2025-03-12 01:03:39.328510: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:03:39.329074: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:03:39.329245: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0\n",
      "2025-03-12 01:03:39.329903: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA\n",
      "2025-03-12 01:03:39.350647: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2499950000 Hz\n",
      "2025-03-12 01:03:39.351660: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55718045b660 executing computations on platform Host. Devices:\n",
      "2025-03-12 01:03:39.351749: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): <undefined>, <undefined>\n",
      "2025-03-12 01:03:39.471585: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:03:39.472948: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x5571817262a0 executing computations on platform CUDA. Devices:\n",
      "2025-03-12 01:03:39.473204: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): NVIDIA GeForce GTX 1650, Compute Capability 7.5\n",
      "2025-03-12 01:03:39.474528: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:03:39.474921: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: \n",
      "name: NVIDIA GeForce GTX 1650 major: 7 minor: 5 memoryClockRate(GHz): 1.515\n",
      "pciBusID: 0000:01:00.0\n",
      "2025-03-12 01:03:39.475078: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
      "2025-03-12 01:03:39.475125: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
      "2025-03-12 01:03:39.475153: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0\n",
      "2025-03-12 01:03:39.475172: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0\n",
      "2025-03-12 01:03:39.475218: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0\n",
      "2025-03-12 01:03:39.475240: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0\n",
      "2025-03-12 01:03:39.475265: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7\n",
      "2025-03-12 01:03:39.475589: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:03:39.476060: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:03:39.476245: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0\n",
      "2025-03-12 01:03:39.476417: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
      "2025-03-12 01:03:39.476901: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1181] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
      "2025-03-12 01:03:39.476924: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1187]      0 \n",
      "2025-03-12 01:03:39.476935: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 0:   N \n",
      "2025-03-12 01:03:39.477412: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:03:39.478360: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:03:39.479158: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1326] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 3545 MB memory) -> physical GPU (device: 0, name: NVIDIA GeForce GTX 1650, pci bus id: 0000:01:00.0, compute capability: 7.5)\n",
      "WARNING:tensorflow:From /home/himanshu/anaconda3/envs/BTP/lib/python3.6/site-packages/tensorflow/python/ops/init_ops.py:1251: calling VarianceScaling.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0d0c0af908>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0d0c0af908>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0d0c001e48>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0d0c001e48>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0d0c0113c8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0d0c0113c8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0d07ff5dd8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0d07ff5dd8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0d001e32b0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0d001e32b0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0d001f27b8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0d001f27b8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0d0c070ba8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0d0c070ba8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0d0015d2b0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0d0015d2b0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0d00175c18>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0d00175c18>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0d00113c88>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0d00113c88>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0d000edba8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0d000edba8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0d0007fe10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0d0007fe10>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:From ./code/model_util.py:98: The name tf.global_variables_initializer is deprecated. Please use tf.compat.v1.global_variables_initializer instead.\n",
      "\n",
      "WARNING:tensorflow:From /home/himanshu/anaconda3/envs/BTP/lib/python3.6/site-packages/tensorflow/python/ops/math_grad.py:1250: add_dispatch_support.<locals>.wrapper (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Use tf.where in 2.0, which has the same broadcast rule as np.where\n",
      "2025-03-12 01:03:43.542258: W tensorflow/compiler/jit/mark_for_compilation_pass.cc:1412] (One-time warning): Not using XLA:CPU for cluster because envvar TF_XLA_FLAGS=--tf_xla_cpu_global_jit was not set.  If you want XLA:CPU, either set that envvar, or use experimental_jit_scope to enable XLA:CPU.  To confirm that XLA is active, pass --vmodule=xla_compilation_cache=1 (as a proper command-line flag, not via TF_XLA_FLAGS) or set the envvar XLA_FLAGS=--xla_hlo_profile.\n",
      "2025-03-12 01:03:43.616773: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
      "2025-03-12 01:03:45.129760: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:03:45.129887: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: \n",
      "name: NVIDIA GeForce GTX 1650 major: 7 minor: 5 memoryClockRate(GHz): 1.515\n",
      "pciBusID: 0000:01:00.0\n",
      "2025-03-12 01:03:45.129928: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
      "2025-03-12 01:03:45.129941: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
      "2025-03-12 01:03:45.129952: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0\n",
      "2025-03-12 01:03:45.129962: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0\n",
      "2025-03-12 01:03:45.129973: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0\n",
      "2025-03-12 01:03:45.129983: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0\n",
      "2025-03-12 01:03:45.129994: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7\n",
      "2025-03-12 01:03:45.130054: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:03:45.130177: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:03:45.130245: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0\n",
      "2025-03-12 01:03:45.130263: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1181] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
      "2025-03-12 01:03:45.130268: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1187]      0 \n",
      "2025-03-12 01:03:45.130272: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 0:   N \n",
      "2025-03-12 01:03:45.130332: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:03:45.130470: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:03:45.130582: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1326] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 3545 MB memory) -> physical GPU (device: 0, name: NVIDIA GeForce GTX 1650, pci bus id: 0000:01:00.0, compute capability: 7.5)\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0b4a32c4a8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0b4a32c4a8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0b4a2e1390>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0b4a2e1390>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0b4a2e29b0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0b4a2e29b0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0b4a2bef28>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0b4a2bef28>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0b4a271390>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0b4a271390>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0b4a20f5f8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0b4a20f5f8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0b4a22d860>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0b4a22d860>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0b4a1e37b8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0b4a1e37b8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0b4a183cf8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0b4a183cf8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0b4a19f8d0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0b4a19f8d0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0b4a164d30>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0b4a164d30>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0b4a177be0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0b4a177be0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "2025-03-12 01:03:54.734161: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:03:54.734281: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: \n",
      "name: NVIDIA GeForce GTX 1650 major: 7 minor: 5 memoryClockRate(GHz): 1.515\n",
      "pciBusID: 0000:01:00.0\n",
      "2025-03-12 01:03:54.734321: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
      "2025-03-12 01:03:54.734335: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
      "2025-03-12 01:03:54.734348: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0\n",
      "2025-03-12 01:03:54.734360: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0\n",
      "2025-03-12 01:03:54.734372: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0\n",
      "2025-03-12 01:03:54.734384: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0\n",
      "2025-03-12 01:03:54.734397: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7\n",
      "2025-03-12 01:03:54.734454: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:03:54.734559: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:03:54.734627: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0\n",
      "2025-03-12 01:03:54.734646: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1181] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
      "2025-03-12 01:03:54.734651: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1187]      0 \n",
      "2025-03-12 01:03:54.734657: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 0:   N \n",
      "2025-03-12 01:03:54.734717: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:03:54.734824: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:03:54.734900: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1326] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 3545 MB memory) -> physical GPU (device: 0, name: NVIDIA GeForce GTX 1650, pci bus id: 0000:01:00.0, compute capability: 7.5)\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0b34832d68>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0b34832d68>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0b347e3e10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0b347e3e10>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0b4a2b94a8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0b4a2b94a8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0b347a8710>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0b347a8710>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0b3477a0b8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0b3477a0b8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0b346fe4e0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0b346fe4e0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0b346cd630>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0b346cd630>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0b346e5240>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0b346e5240>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0b346a3fd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0b346a3fd0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0b3464a748>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0b3464a748>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0b34669e80>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f0b34669e80>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0b34604d68>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f0b34604d68>>: AttributeError: module 'gast' has no attribute 'Index'\n"
     ]
    }
   ],
   "source": [
    "!bash run.sbatch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-18 0.502204206503372 -9 0.2224660045104505\n",
      "-17.397747\n",
      "Figure(640x480)\n",
      "148\n"
     ]
    }
   ],
   "source": [
    "!python exact.py"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![alacpd](./results_AE_skipLSTM_AR/std/seed=0/log/3.png)\n",
    "![exact](./exact_fig/3.png)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[160]\n"
     ]
    }
   ],
   "source": [
    "!cd datasets/std && python create.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+ model_name=AE_skipLSTM_AR\n",
      "+ seed=0\n",
      "+ windows=6\n",
      "+ train_percent=0.1\n",
      "+ dataset=std\n",
      "+ e_init=10\n",
      "+ e_train=5\n",
      "+ e_reinit=100\n",
      "+ n_cpd=10\n",
      "+ unit=20\n",
      "+ horizon=4\n",
      "+ highway=6\n",
      "+ skip_sizes='3 5 7'\n",
      "+ python3 -W ignore ./code/main.py --dataset_name std --seed 0 --windows 6 --train_percent 0.1 --model_name AE_skipLSTM_AR --unit 20 --horizon 4 --highway 6 --skip_sizes 3 5 7 --e_init 10 --e_train 5 --e_reinit 100 --n_cpd 10\n",
      "Namespace(dataset_name='std', e_init=10, e_reinit=100, e_train=5, highway=6, horizon=4, model_name='AE_skipLSTM_AR', n_cpd=10, seed=0, skip_sizes=[3, 5, 7], train_percent=0.1, unit=20, windows=6)\n",
      "{'dataset_name': 'std', 'skip_sizes': [3, 5, 7], 'train_percent': 0.1, 'training_epochs': 10, 'horizon': 4, 'highway': 6, 'model_names': ['AE_skipLSTM_AR', 'AE_skipLSTM_AR', 'AE_skipLSTM_AR'], 'lr': 0.001, 'GRUUnits': 20, 'SkipGRUUnits': 20, 'window': 6, 'num_change_threshold': 7, 'num_ano_cpd': 10, 'threshold_high': 4, 'threshold_low': 1.4, 'epochs_to_train_after_cpd': 100, 'extra_samples_after_cpd': 3, 'epochs_to_train_single_sample': 5, 'file_name': './results_AE_skipLSTM_AR/std/seed=0/', 'ensemble_space': 3}\n",
      "WARNING:tensorflow:From ./code/main.py:40: The name tf.random.set_random_seed is deprecated. Please use tf.compat.v1.random.set_random_seed instead.\n",
      "\n",
      "./datasets/std/std.json\n",
      "Log path  ./results_AE_skipLSTM_AR/std/seed=0/log.out\n",
      "2025-03-12 01:10:21.757574: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcuda.so.1\n",
      "2025-03-12 01:10:21.791718: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:10:21.792424: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: \n",
      "name: NVIDIA GeForce GTX 1650 major: 7 minor: 5 memoryClockRate(GHz): 1.515\n",
      "pciBusID: 0000:01:00.0\n",
      "2025-03-12 01:10:21.793589: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
      "2025-03-12 01:10:21.796372: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
      "2025-03-12 01:10:21.798687: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0\n",
      "2025-03-12 01:10:21.801068: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0\n",
      "2025-03-12 01:10:21.803203: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0\n",
      "2025-03-12 01:10:21.805568: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0\n",
      "2025-03-12 01:10:21.813211: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7\n",
      "2025-03-12 01:10:21.813468: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:10:21.814032: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:10:21.814239: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0\n",
      "2025-03-12 01:10:21.814640: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA\n",
      "2025-03-12 01:10:21.839990: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2499950000 Hz\n",
      "2025-03-12 01:10:21.841247: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x555e6dc97660 executing computations on platform Host. Devices:\n",
      "2025-03-12 01:10:21.841347: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): <undefined>, <undefined>\n",
      "2025-03-12 01:10:21.975248: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:10:21.975778: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x555e6ef62450 executing computations on platform CUDA. Devices:\n",
      "2025-03-12 01:10:21.975885: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): NVIDIA GeForce GTX 1650, Compute Capability 7.5\n",
      "2025-03-12 01:10:21.977235: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:10:21.977780: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: \n",
      "name: NVIDIA GeForce GTX 1650 major: 7 minor: 5 memoryClockRate(GHz): 1.515\n",
      "pciBusID: 0000:01:00.0\n",
      "2025-03-12 01:10:21.978072: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
      "2025-03-12 01:10:21.978130: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
      "2025-03-12 01:10:21.978152: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0\n",
      "2025-03-12 01:10:21.978165: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0\n",
      "2025-03-12 01:10:21.978176: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0\n",
      "2025-03-12 01:10:21.978188: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0\n",
      "2025-03-12 01:10:21.978199: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7\n",
      "2025-03-12 01:10:21.978445: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:10:21.978907: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:10:21.979046: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0\n",
      "2025-03-12 01:10:21.979120: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
      "2025-03-12 01:10:21.980322: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1181] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
      "2025-03-12 01:10:21.980475: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1187]      0 \n",
      "2025-03-12 01:10:21.980493: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 0:   N \n",
      "2025-03-12 01:10:21.981335: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:10:21.981970: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:10:21.982135: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1326] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 3545 MB memory) -> physical GPU (device: 0, name: NVIDIA GeForce GTX 1650, pci bus id: 0000:01:00.0, compute capability: 7.5)\n",
      "WARNING:tensorflow:From /home/himanshu/anaconda3/envs/BTP/lib/python3.6/site-packages/tensorflow/python/ops/init_ops.py:1251: calling VarianceScaling.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8ae004d908>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8ae004d908>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f8adbf81e48>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f8adbf81e48>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8adbf923c8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8adbf923c8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f8adbf75dd8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f8adbf75dd8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8adbf2b2b0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8adbf2b2b0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f8adbf397b8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f8adbf397b8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8adbe86ba8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8adbe86ba8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f8adbea22b0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f8adbea22b0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8adbebdc18>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8adbebdc18>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f8ad41dbc88>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f8ad41dbc88>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8adbeedba8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8adbeedba8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f8ad41b0b38>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f8ad41b0b38>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:From ./code/model_util.py:98: The name tf.global_variables_initializer is deprecated. Please use tf.compat.v1.global_variables_initializer instead.\n",
      "\n",
      "WARNING:tensorflow:From /home/himanshu/anaconda3/envs/BTP/lib/python3.6/site-packages/tensorflow/python/ops/math_grad.py:1250: add_dispatch_support.<locals>.wrapper (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Use tf.where in 2.0, which has the same broadcast rule as np.where\n",
      "2025-03-12 01:10:27.550716: W tensorflow/compiler/jit/mark_for_compilation_pass.cc:1412] (One-time warning): Not using XLA:CPU for cluster because envvar TF_XLA_FLAGS=--tf_xla_cpu_global_jit was not set.  If you want XLA:CPU, either set that envvar, or use experimental_jit_scope to enable XLA:CPU.  To confirm that XLA is active, pass --vmodule=xla_compilation_cache=1 (as a proper command-line flag, not via TF_XLA_FLAGS) or set the envvar XLA_FLAGS=--xla_hlo_profile.\n",
      "2025-03-12 01:10:27.624609: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
      "2025-03-12 01:10:29.067049: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:10:29.067172: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: \n",
      "name: NVIDIA GeForce GTX 1650 major: 7 minor: 5 memoryClockRate(GHz): 1.515\n",
      "pciBusID: 0000:01:00.0\n",
      "2025-03-12 01:10:29.067212: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
      "2025-03-12 01:10:29.067224: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
      "2025-03-12 01:10:29.067249: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0\n",
      "2025-03-12 01:10:29.067260: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0\n",
      "2025-03-12 01:10:29.067271: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0\n",
      "2025-03-12 01:10:29.067282: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0\n",
      "2025-03-12 01:10:29.067293: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7\n",
      "2025-03-12 01:10:29.067350: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:10:29.067458: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:10:29.067528: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0\n",
      "2025-03-12 01:10:29.067546: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1181] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
      "2025-03-12 01:10:29.067551: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1187]      0 \n",
      "2025-03-12 01:10:29.067555: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 0:   N \n",
      "2025-03-12 01:10:29.067616: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:10:29.067739: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:10:29.067820: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1326] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 3545 MB memory) -> physical GPU (device: 0, name: NVIDIA GeForce GTX 1650, pci bus id: 0000:01:00.0, compute capability: 7.5)\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8a444624a8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8a444624a8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f8a4441f390>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f8a4441f390>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8a444189b0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8a444189b0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f8a443f5f28>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f8a443f5f28>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8a443a9390>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8a443a9390>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f8a443475f8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f8a443475f8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8a44366860>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8a44366860>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f8a4431a780>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f8a4431a780>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8a442bbcc0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8a442bbcc0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f8a442d7c88>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f8a442d7c88>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8a4429b5f8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8a4429b5f8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f8a442afbe0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f8a442afbe0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "2025-03-12 01:10:33.170123: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:10:33.170245: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: \n",
      "name: NVIDIA GeForce GTX 1650 major: 7 minor: 5 memoryClockRate(GHz): 1.515\n",
      "pciBusID: 0000:01:00.0\n",
      "2025-03-12 01:10:33.170285: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
      "2025-03-12 01:10:33.170298: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
      "2025-03-12 01:10:33.170308: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0\n",
      "2025-03-12 01:10:33.170319: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0\n",
      "2025-03-12 01:10:33.170329: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0\n",
      "2025-03-12 01:10:33.170340: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0\n",
      "2025-03-12 01:10:33.170351: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7\n",
      "2025-03-12 01:10:33.170407: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:10:33.170513: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:10:33.170583: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0\n",
      "2025-03-12 01:10:33.170599: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1181] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
      "2025-03-12 01:10:33.170604: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1187]      0 \n",
      "2025-03-12 01:10:33.170609: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 0:   N \n",
      "2025-03-12 01:10:33.170668: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:10:33.170775: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:10:33.170851: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1326] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 3545 MB memory) -> physical GPU (device: 0, name: NVIDIA GeForce GTX 1650, pci bus id: 0000:01:00.0, compute capability: 7.5)\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f88ebef6d68>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f88ebef6d68>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f88ebea7e10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f88ebea7e10>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8a443f04a8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f8a443f04a8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f88ebe6c710>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f88ebe6c710>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f88ebe3f0b8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f88ebe3f0b8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f88ebdc24e0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f88ebdc24e0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f88ebd93f60>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f88ebd93f60>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f88ebdaa320>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f88ebdaa320>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f88ebd67860>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f88ebd67860>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f88ebd13208>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f88ebd13208>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f88ebd2eda0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f88ebd2eda0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f88ebccad68>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f88ebccad68>>: AttributeError: module 'gast' has no attribute 'Index'\n"
     ]
    }
   ],
   "source": [
    "!bash run.sbatch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-5 0.1575710432602747 -1 0.8685223758304617\n",
      "-5.053662\n",
      "Figure(640x480)\n",
      "160\n"
     ]
    }
   ],
   "source": [
    "!python exact.py"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![alacpd](./results_AE_skipLSTM_AR/std/seed=0/log/4.png)\n",
    "![exact](./exact_fig/4.png)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[142]\n"
     ]
    }
   ],
   "source": [
    "!cd datasets/std && python create.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+ model_name=AE_skipLSTM_AR\n",
      "+ seed=0\n",
      "+ windows=6\n",
      "+ train_percent=0.1\n",
      "+ dataset=std\n",
      "+ e_init=10\n",
      "+ e_train=5\n",
      "+ e_reinit=100\n",
      "+ n_cpd=10\n",
      "+ unit=20\n",
      "+ horizon=4\n",
      "+ highway=6\n",
      "+ skip_sizes='3 5 7'\n",
      "+ python3 -W ignore ./code/main.py --dataset_name std --seed 0 --windows 6 --train_percent 0.1 --model_name AE_skipLSTM_AR --unit 20 --horizon 4 --highway 6 --skip_sizes 3 5 7 --e_init 10 --e_train 5 --e_reinit 100 --n_cpd 10\n",
      "Namespace(dataset_name='std', e_init=10, e_reinit=100, e_train=5, highway=6, horizon=4, model_name='AE_skipLSTM_AR', n_cpd=10, seed=0, skip_sizes=[3, 5, 7], train_percent=0.1, unit=20, windows=6)\n",
      "{'dataset_name': 'std', 'skip_sizes': [3, 5, 7], 'train_percent': 0.1, 'training_epochs': 10, 'horizon': 4, 'highway': 6, 'model_names': ['AE_skipLSTM_AR', 'AE_skipLSTM_AR', 'AE_skipLSTM_AR'], 'lr': 0.001, 'GRUUnits': 20, 'SkipGRUUnits': 20, 'window': 6, 'num_change_threshold': 7, 'num_ano_cpd': 10, 'threshold_high': 4, 'threshold_low': 1.4, 'epochs_to_train_after_cpd': 100, 'extra_samples_after_cpd': 3, 'epochs_to_train_single_sample': 5, 'file_name': './results_AE_skipLSTM_AR/std/seed=0/', 'ensemble_space': 3}\n",
      "WARNING:tensorflow:From ./code/main.py:40: The name tf.random.set_random_seed is deprecated. Please use tf.compat.v1.random.set_random_seed instead.\n",
      "\n",
      "./datasets/std/std.json\n",
      "Log path  ./results_AE_skipLSTM_AR/std/seed=0/log.out\n",
      "2025-03-12 01:15:07.689710: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcuda.so.1\n",
      "2025-03-12 01:15:07.730507: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:15:07.731038: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: \n",
      "name: NVIDIA GeForce GTX 1650 major: 7 minor: 5 memoryClockRate(GHz): 1.515\n",
      "pciBusID: 0000:01:00.0\n",
      "2025-03-12 01:15:07.731334: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
      "2025-03-12 01:15:07.734639: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
      "2025-03-12 01:15:07.739053: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0\n",
      "2025-03-12 01:15:07.740073: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0\n",
      "2025-03-12 01:15:07.745687: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0\n",
      "2025-03-12 01:15:07.748714: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0\n",
      "2025-03-12 01:15:07.758316: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7\n",
      "2025-03-12 01:15:07.758655: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:15:07.759029: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:15:07.759351: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0\n",
      "2025-03-12 01:15:07.760470: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA\n",
      "2025-03-12 01:15:07.787764: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2499950000 Hz\n",
      "2025-03-12 01:15:07.789143: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x563217357660 executing computations on platform Host. Devices:\n",
      "2025-03-12 01:15:07.789297: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): <undefined>, <undefined>\n",
      "2025-03-12 01:15:07.954298: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:15:07.954628: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x563218622080 executing computations on platform CUDA. Devices:\n",
      "2025-03-12 01:15:07.954648: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): NVIDIA GeForce GTX 1650, Compute Capability 7.5\n",
      "2025-03-12 01:15:07.955110: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:15:07.955456: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: \n",
      "name: NVIDIA GeForce GTX 1650 major: 7 minor: 5 memoryClockRate(GHz): 1.515\n",
      "pciBusID: 0000:01:00.0\n",
      "2025-03-12 01:15:07.955687: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
      "2025-03-12 01:15:07.955755: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
      "2025-03-12 01:15:07.955810: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0\n",
      "2025-03-12 01:15:07.955941: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0\n",
      "2025-03-12 01:15:07.956029: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0\n",
      "2025-03-12 01:15:07.956099: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0\n",
      "2025-03-12 01:15:07.956169: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7\n",
      "2025-03-12 01:15:07.957021: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:15:07.957421: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:15:07.957624: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0\n",
      "2025-03-12 01:15:07.957948: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
      "2025-03-12 01:15:07.958742: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1181] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
      "2025-03-12 01:15:07.958784: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1187]      0 \n",
      "2025-03-12 01:15:07.958822: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 0:   N \n",
      "2025-03-12 01:15:07.959225: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:15:07.959528: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:15:07.959762: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1326] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 3545 MB memory) -> physical GPU (device: 0, name: NVIDIA GeForce GTX 1650, pci bus id: 0000:01:00.0, compute capability: 7.5)\n",
      "WARNING:tensorflow:From /home/himanshu/anaconda3/envs/BTP/lib/python3.6/site-packages/tensorflow/python/ops/init_ops.py:1251: calling VarianceScaling.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f950bfae390>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f950bfae390>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f950bf1a2e8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f950bf1a2e8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f950bf08390>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f950bf08390>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f950bef5320>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f950bef5320>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f950beab2b0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f950beab2b0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f950beb9860>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f950beb9860>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f9504186ba8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f9504186ba8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f95041a2470>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f95041a2470>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f95041bdc18>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f95041bdc18>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f950415bba8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f950415bba8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f95041efba8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f95041efba8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f950412fbe0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f950412fbe0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:From ./code/model_util.py:98: The name tf.global_variables_initializer is deprecated. Please use tf.compat.v1.global_variables_initializer instead.\n",
      "\n",
      "WARNING:tensorflow:From /home/himanshu/anaconda3/envs/BTP/lib/python3.6/site-packages/tensorflow/python/ops/math_grad.py:1250: add_dispatch_support.<locals>.wrapper (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Use tf.where in 2.0, which has the same broadcast rule as np.where\n",
      "2025-03-12 01:15:20.669249: W tensorflow/compiler/jit/mark_for_compilation_pass.cc:1412] (One-time warning): Not using XLA:CPU for cluster because envvar TF_XLA_FLAGS=--tf_xla_cpu_global_jit was not set.  If you want XLA:CPU, either set that envvar, or use experimental_jit_scope to enable XLA:CPU.  To confirm that XLA is active, pass --vmodule=xla_compilation_cache=1 (as a proper command-line flag, not via TF_XLA_FLAGS) or set the envvar XLA_FLAGS=--xla_hlo_profile.\n",
      "2025-03-12 01:15:21.012030: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
      "2025-03-12 01:15:25.418208: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:15:25.418383: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: \n",
      "name: NVIDIA GeForce GTX 1650 major: 7 minor: 5 memoryClockRate(GHz): 1.515\n",
      "pciBusID: 0000:01:00.0\n",
      "2025-03-12 01:15:25.418438: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
      "2025-03-12 01:15:25.418452: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
      "2025-03-12 01:15:25.418464: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0\n",
      "2025-03-12 01:15:25.418477: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0\n",
      "2025-03-12 01:15:25.418488: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0\n",
      "2025-03-12 01:15:25.418500: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0\n",
      "2025-03-12 01:15:25.418511: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7\n",
      "2025-03-12 01:15:25.418586: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:15:25.418730: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:15:25.418804: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0\n",
      "2025-03-12 01:15:25.418828: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1181] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
      "2025-03-12 01:15:25.418833: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1187]      0 \n",
      "2025-03-12 01:15:25.418838: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 0:   N \n",
      "2025-03-12 01:15:25.418909: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:15:25.419032: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:15:25.419120: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1326] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 3545 MB memory) -> physical GPU (device: 0, name: NVIDIA GeForce GTX 1650, pci bus id: 0000:01:00.0, compute capability: 7.5)\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f94b0562748>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f94b0562748>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f94b051f3c8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f94b051f3c8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f94b0518940>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f94b0518940>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f94b04f5f28>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f94b04f5f28>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f94b04a9080>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f94b04a9080>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f94b0439908>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f94b0439908>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f94b0466860>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f94b0466860>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f94b041a9e8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f94b041a9e8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f94b03bacf8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f94b03bacf8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f94b03d7cc0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f94b03d7cc0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f94b039b5f8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f94b039b5f8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f94b03aff60>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f94b03aff60>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "2025-03-12 01:15:31.007780: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:15:31.007911: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: \n",
      "name: NVIDIA GeForce GTX 1650 major: 7 minor: 5 memoryClockRate(GHz): 1.515\n",
      "pciBusID: 0000:01:00.0\n",
      "2025-03-12 01:15:31.007953: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
      "2025-03-12 01:15:31.007968: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
      "2025-03-12 01:15:31.007982: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0\n",
      "2025-03-12 01:15:31.007995: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0\n",
      "2025-03-12 01:15:31.008009: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0\n",
      "2025-03-12 01:15:31.008022: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0\n",
      "2025-03-12 01:15:31.008035: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7\n",
      "2025-03-12 01:15:31.008097: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:15:31.008213: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:15:31.008286: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0\n",
      "2025-03-12 01:15:31.008306: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1181] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
      "2025-03-12 01:15:31.008312: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1187]      0 \n",
      "2025-03-12 01:15:31.008318: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 0:   N \n",
      "2025-03-12 01:15:31.008383: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:15:31.008501: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2025-03-12 01:15:31.008595: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1326] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 3545 MB memory) -> physical GPU (device: 0, name: NVIDIA GeForce GTX 1650, pci bus id: 0000:01:00.0, compute capability: 7.5)\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f94681efd30>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f94681efd30>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f94681a0d30>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f94681a0d30>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f94b04ef5c0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f94b04ef5c0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f9468165a90>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f9468165a90>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f94681380b8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f94681380b8>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f94680bb588>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f94680bb588>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f946808cf60>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f946808cf60>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f94680a3320>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f94680a3320>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f9468060860>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f9468060860>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f944c7cc1d0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f944c7cc1d0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f944c7e7da0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PreARTrans.call of <lstmae_ensemble.PreARTrans object at 0x7f944c7e7da0>>: AttributeError: module 'gast' has no attribute 'Index'\n",
      "WARNING:tensorflow:Entity <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f944c783cf8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method PostARTrans.call of <lstmae_ensemble.PostARTrans object at 0x7f944c783cf8>>: AttributeError: module 'gast' has no attribute 'Index'\n"
     ]
    }
   ],
   "source": [
    "!bash run.sbatch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-12 0.6026400568526976 -10 0.6833632073273981\n",
      "-12.747449\n",
      "Figure(640x480)\n",
      "142\n"
     ]
    }
   ],
   "source": [
    "!python exact.py"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![alacpd](./results_AE_skipLSTM_AR/std/seed=0/log/5.png)\n",
    "![exact](./exact_fig/5.png)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "BTP",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
